### PR TITLE
changing csv-import of map to include pk in struct

### PR DESCRIPTION
**Pull request by mikegray**
fixing issue #2292

### CODE REVIEW

File: samples/go/csv/read.go [hunk 1]
```diff
@@ -146,17 +131,18 @@ func ReadToMap(r *csv.Reader, headersRaw []string, pkIdx int, kinds KindSlice, v
 
 		fieldIndex := 0
 		var pk types.Value
-		fields := make(types.ValueSlice, len(headers))
+		fields := make(types.ValueSlice, len(headersRaw))
 		for x, v := range row {
-			if x == pkIdx {
-				pk, err = StringToValue(v, pkKind)
-			} else if fieldIndex < len(headers) {
+			if fieldIndex < len(headersRaw) {
```
Comment (aboodman) [inline samples/go/csv/read.go]: Can `fieldIndex` ever be less than `len(headersRaw)` now?

Comment (mikegray) [inline samples/go/csv/read.go]: should always be less, except if the csv file has a different number of fields that the header row indicated.

Comment (willhite) [inline samples/go/csv/read.go]: It looks like we're ignoring any fields that exceed the number of headers. Shouldn't we print a warning or something? Also, I don't see why we need the fieldIndex var here. Can't you just use 'x'?

Comment (mikegray) [inline samples/go/csv/read.go]: fixed

**Comment by mikegray** [comment]
PTAL again

**Comment by willhite** [comment]
LGTM. One point though, In the past, we were ignoring the case where there were more fields in a row than headers which means that we will now fail on .csv files that we were able to process. If that's the case we might need to turn the error for this into a warning.

**Comment by mikegray** [comment]
@aboodman what is your thought here - do we panic and stop parsing? just give a warning? or do we silently ignore the issue and not import those fields (this was the default before)?

**Comment by aboodman** [comment]
We actually rely on the old behavior of silent ignore, so we need to keep it.

**Comment by mikegray** [comment]
Ok @willhite I went back to the original notion of ignoring extra fields. PTAL again, minor changes since last review.

**Pull request merged**